### PR TITLE
Run nginx in a docker container and serve static files

### PR DESCRIPTION
**Pull request by jhamrick**
This is largely taken from the tmpnb config, see https://github.com/jupyter/tmpnb-deploy/pull/3

This seems to work ok but could you take a look at it too @rgbkrk ?

### CODE REVIEW

**Comment by rgbkrk** [comment]
Hopefully you saw the final version too, not just jupyter/tmpnb-deploy#3 as there may have been changes after. :wink: I'll check it out.

File: roles/proxy/tasks/main.yml [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+
+# Get static resources from the userland container
+- name: static files container
+  docker:
+    image: jupyter/systemuser
+    state: running
+    name: static-files
+    command: echo "#data-container-hack"
+    volumes:
+    - "{{ notebook_static_files }}"
```
Comment (rgbkrk) [inline roles/proxy/tasks/main.yml:14]: Just a note that when `jupyter/systemuser` changes, we need to make sure what nginx has for static resources matches user land containers.

**Comment by rgbkrk** [comment]
Looks great on my initial read through! Appears you made it fully up to date too.

File: roles/proxy/tasks/main.yml (hunk 1 above)
Comment (jhamrick) [inline roles/proxy/tasks/main.yml:14]: How do we ensure that? Just pulling the new version of jupyter/systemuser on the proxy, and restarting nginx? Or were you thinking of something else?

Comment (rgbkrk) [inline roles/proxy/tasks/main.yml:14]: Yeah, totally. Just a side note. If they're ever out of sync, IPython may not behave well for clients.

Comment (jhamrick) [inline roles/proxy/tasks/main.yml:14]: Ok, noted!

**Pull request merged**